python/taichi/ui/scene.py

```diff
@@ -125,10 +127,13 @@ def particles(self,
         vbo = get_vbo_field(centers)
         copy_vertices_to_vbo(vbo, centers)
         has_per_vertex_color = per_vertex_color is not None
+        has_per_vertex_radius = per_vertex_radius is not None
         if has_per_vertex_color:
             copy_colors_to_vbo(vbo, per_vertex_color)
+        if has_per_vertex_radius:
+            copy_texcoords_to_vbo(vbo, per_vertex_radius)
```

Comment:
`copy_texcoords_to_vbo` won't work out of the box here. The `per_vertex_radius` field provided by the user should be a scalar field, but `copy_texcoords_to_vbo` expects a 2D vector field. You'll need to create a new function: something like `copy_per_vertex_radius_to_vbo`.